src/AutoUpdate/Hooks.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace WPML\PB\AutoUpdate;
+
+use WPML\FP\Logic;
+use WPML\FP\Lst;
+use WPML\FP\Obj;
+use WPML\FP\Relation;
+use WPML\FP\Str;
+use function WPML\FP\invoke;
+use function WPML\FP\pipe;
+
+class Hooks implements \IWPML_Backend_Action, \IWPML_Frontend_Action, \IWPML_DIC_Action {
+
+	const HASH_SEP = '-';
+
+	/** @var \WPML_PB_Integration $pbIntegration */
+	private $pbIntegration;
+
+	/** @var \WPML_Translation_Element_Factory $elementFactory */
+	private $elementFactory;
+
+	public function __construct(
+		\WPML_PB_Integration $pbIntegration,
+		\WPML_Translation_Element_Factory $elementFactory
+	) {
+		$this->pbIntegration     = $pbIntegration;
+		$this->elementFactory    = $elementFactory;
+	}
+
+	public function add_hooks() {
+		add_filter( 'wpml_tm_post_md5_content', [ $this, 'getMd5ContentFromPackageStrings' ], 10, 2 );
+		add_action( 'wpml_after_save_post', [ $this, 'resaveTranslationsAfterSavePost' ], 10, 4 );
+	}
+
+	/**
+	 * @param string   $content
+	 * @param \WP_Post $post
+	 *
+	 * @return string
+	 */
+	public function getMd5ContentFromPackageStrings( $content, $post ) {
+		if ( ! self::getPackages( $post->ID ) ) {
+			return $content;
+		}
+
+		// $joinPackageStringHashes :: \WPML_Package → string
+		$joinPackageStringHashes = pipe(
+			Obj::propOr( [], 'string_data' ),
+			'array_keys',
```

Comment:
@strategio I also defined recently: `Obj::keys`. You can use it if you wish.

Reply:
Right @gordonsh24, applied your suggestion!